### DISCUSSION TITLE
[Camera Support]: Dahua PTZ No hope for autotracking?

### DISCUSSION THREAD

**Discussion by jakeycrx** [blakeblackshear/frigate#12194] [Camera Support] [answered]
### Describe the problem you are having

Hi,

I just picked up a Dahua SD49412T-HN. It's an older discontinued model, but seems to have a pretty good spec.  I bought it specifically to autotrack on my driveway, but it looks like it's not going to happen.  

This is the output from fovtest.py, is there no hope? I can't find any newer firmware for it anywhere.

YES - GetServiceCapabilities shows that the camera supports MoveStatus.
YES - MoveStatus is reporting IDLE.
NO - RelativeMove Pan/Tilt is unsupported.
YES - RelativeMove Zoom is supported.

### Version

0.13.2

### Frigate config file

```yaml
N/A
```

### Relevant log output

```shell
N/A
```

### FFprobe output from your camera

```shell
N/A
```

### Operating system

HassOS

### Install method

HassOS Addon

### Object Detector

OpenVino

### Network connection

Wired

### Camera make and model

Dahua DH-SD49412T-HN

**Comment by hawkeye217** [comment]
No hope, unfortunately. Relative movement is required.

**Reply by jakeycrx** [reply]
Damn it I got it for a good price, but it was still a bit of a spend.  It looks like Dahua have never updated the firmware after the initial release, it's dated 2018.  I'll email the help desk and see what they say.  

The FOV is quite narrow on it, so the view is fairly restricted even though it's mounted up high.

**Reply by hawkeye217** [reply]
🤞

**Reply by jakeycrx** [reply]
I was just looking at the log and it says relative zoom isn't supported?  The fovtest.py output says RelativeMove Zoom is supported though, or am I reading that wrong?

2024-06-29 20:56:45.027084837  [2024-06-29 21:56:45] frigate.ptz.onvif              WARNING : Disabling autotracking zooming for dahua: Relative zoom not supported
2024-06-29 20:56:45.165434302  [2024-06-29 21:56:45] frigate.ptz.autotrack          WARNING : Disabling autotracking for dahua: FOV relative movement not supported

I looked up the section for RelativePanTiltTranslationSpace and it does list min and max values? 

        'RelativePanTiltTranslationSpace': [
            {
                'URI': 'http://www.onvif.org/ver10/tptz/PanTiltSpaces/TranslationGenericSpace',
                'XRange': {
                    'Min': -1.0,
                    'Max': 1.0
                },
                'YRange': {
                    'Min': -1.0,
                    'Max': 1.0

Here is the full output if you want to take a look

https://pastebin.com/DKt6ZUFe

**Reply by hawkeye217** [reply]
There's a specific relative pan/tilt translation space in the ONVIF standard called `TranslationSpaceFov` that your camera lacks the support for. 

For example, my supported camera shows this:

```
frigate    | 2023-11-16 16:19:01.843229816          'RelativePanTiltTranslationSpace': [
frigate    | 2023-11-16 16:19:01.843234383              {
frigate    | 2023-11-16 16:19:01.843241407                  'URI': 'http://www.onvif.org/ver10/tptz/PanTiltSpaces/TranslationGenericSpace',
frigate    | 2023-11-16 16:19:01.843246540                  'XRange': {
frigate    | 2023-11-16 16:19:01.843251610                      'Min': -1.0,
frigate    | 2023-11-16 16:19:01.843256507                      'Max': 1.0
frigate    | 2023-11-16 16:19:01.843261080                  },
frigate    | 2023-11-16 16:19:01.843266040                  'YRange': {
frigate    | 2023-11-16 16:19:01.843271247                      'Min': -1.0,
frigate    | 2023-11-16 16:19:01.843276417                      'Max': 1.0
frigate    | 2023-11-16 16:19:01.843281077                  }
frigate    | 2023-11-16 16:19:01.843285914              },
frigate    | 2023-11-16 16:19:01.843290741              {
frigate    | 2023-11-16 16:19:01.843297434                  'URI': 'http://www.onvif.org/ver10/tptz/PanTiltSpaces/TranslationSpaceFov',
frigate    | 2023-11-16 16:19:01.843302171                  'XRange': {
frigate    | 2023-11-16 16:19:01.843307341                      'Min': -1.0,
frigate    | 2023-11-16 16:19:01.843312364                      'Max': 1.0
frigate    | 2023-11-16 16:19:01.843317021                  },
frigate    | 2023-11-16 16:19:01.843321998                  'YRange': {
frigate    | 2023-11-16 16:19:01.843327124                      'Min': -1.0,
frigate    | 2023-11-16 16:19:01.843332131                      'Max': 1.0
frigate    | 2023-11-16 16:19:01.843336978                  }
frigate    | 2023-11-16 16:19:01.843341702              }
frigate    | 2023-11-16 16:19:01.843346269          ],
```

**Reply by jakeycrx** [reply]
Thanks for the explanation.  Which model is that?

**Reply by hawkeye217** [reply]
It's the Dahua SD1A404XB-GNR mini PTZ.

**Reply by jakeycrx** [reply]
It looks a good little camera, nice wide FOV as well. Mine maxes out at 58 degrees which means it's very limited without autotracking.  I spent hours hanging off the side of the roof to mount it as high as possible as well. I might just stick my camera on eBay and switch to your camera as I know it definitely works. It uses the same PFB305W mounting bracket as mine, so would be an easy swap.

**Reply by hawkeye217** [reply]
It's a nice little camera for sure. Only 4x zoom, but you can't really expect much more than that from such a small size.

I bought mine from EmpireTech (Dahua rebrand) and it looks like the model # changed to [PTZ1A4M-4X-S2](https://empiretech01.com/collections/vendors?q=PTZ1A4M-4X-S2).

I also tried the [PTZ425DB-AT](https://empiretech01.com/collections/vendors?q=PTZ425DB-AT) before I settled on the mini PTZ. It was just too big for the space.

**Reply by jakeycrx** [reply]
I just came across this, I'm tempted! It has a nice wide FOV on the panoramic camera.

https://www.dahuasecurity.com/products/All-Products/PTZ-Cameras/X-Spans-Series/SDT4E425-4F-GB-A-PV1

After looking at various other Dahua PTZ models, it seems that the listing of autotracking in the spec means they should support TranslationSpaceFov.  Do you think that's a correct assumption?  This is from the model I linked above:

![image](https://github.com/blakeblackshear/frigate/assets/47118674/7a9038c5-89d4-4899-91bf-10d25f4584d3)

This is the PTZ425DB-AT:

![image](https://github.com/blakeblackshear/frigate/assets/47118674/416e2cea-0e99-40b5-80ad-75ff272f50f7)

Your SD1A404XB-GNR:

![image](https://github.com/blakeblackshear/frigate/assets/47118674/241fdfb9-9fe1-4202-a958-9b51017d451b)

This is my current camera.  I wish I'd picked up on that before I ordered it! 

![image](https://github.com/blakeblackshear/frigate/assets/47118674/03edff68-550a-4720-b590-970427c43dc0)

**Comment by hawkeye217** [comment]
Autotracking provided by the camera firmware is going to outperform Frigate's autotracking every time. It may only track people though. Frigate made it easy for me to write a "poor man's autotracking" using ONVIF for cameras that support the FOV translation space.

Autotracking in firmware gives no indication of whether or not the appropriate ONVIF space is supported, unfortunately.

**Reply by jakeycrx** [reply]
Can you run the OE autotracking with Frigate?

**Reply by hawkeye217** [reply]
If you want to use the onboard autotracking and configure Frigate for recordings only (and not object detection), then it should work great. But if you want to use object detection, a moving camera will confuse Frigate and you'll end up with lots of different events/review items for the same object.

**Reply by jakeycrx** [reply, marked as answer]
After lots of Googling about Dahua firmware upgrades I came across the IPCamTalk forum.  People mentioned Andy at EmpireTech is the man to speak to about Dahua firmware upgrades.  I dropped him a message and he pointed me towards some firmware on his website. This is what the upgraded firmware resulted in:

YES - GetServiceCapabilities shows that the camera supports MoveStatus.
YES - MoveStatus is reporting IDLE.
YES - RelativeMove Pan/Tilt (FOV) is supported.
YES - RelativeMove Zoom is supported.

I now have autotracking! 

![carlton-dance](https://github.com/blakeblackshear/frigate/assets/47118674/1481b1f0-137b-414c-a7b4-85b16d29d886)

**Reply by hawkeye217** [reply]
Great! Andy is awesome. I've bought a number of cams from him.

**Reply by jakeycrx** [reply]
I've just been testing it by walking around and sneaking directly underneath the camera and I'm very impressed! One thing I have noticed is that it creates multiple events, is that to be expected?

**Reply by hawkeye217** [reply]
Yes, but this is greatly improved in the next version of Frigate.

**Comment by dz4news** [comment]
@hawkeye217  I just picked up the PTZ425DB-AT from EmpireTech... you mentioned earlier in the thread that you had tried this PTZ before settling on the mini. I am assuming the larger one worked ok in Frigate?

**Reply by hawkeye217** [reply]
I never tried it with Frigate, actually. Though I can't imagine why it wouldn't support FOV relative movement.

**Reply by dz4news** [reply]
Thanks... I will report back once I get it installed.. I just powered it up and added it to the config, it seems to be working. I will know more tomorrow.

**Reply by hawkeye217** [reply]
The onboard autotracking on that model is great. Much better than anything Frigate could ever do with ONVIF.

**Reply by dz4news** [reply]
I really want it for the 25x zoom.. my entrance is a considerable distance from my house, so I want to reach out see who is there.. the 4x just wasnt doing it. The autotracking will just be icing on the cake.
I appreciate your insight about the camera..

**Reply by hawkeye217** [reply]
You'll love the zoom, then. It was too powerful for the space I had it in.

**Reply by dz4news** [reply]
Now that you have me thinking about this... can I have the best of both? Is it possible to use the onboard to track humans and vehicles and then use Frigate to track such things as dogs and deer that are in my frontyard zone?
Secondly, if the above is possible will frigate still record the activity... if I set to active object or motion?

**Reply by hawkeye217** [reply]
When Frigate doesn't control the autotracking, you may end up with multiple tracked objects due to the motion of the camera. It may not be ideal, but with the right recording retention, you may be fine with that.

**Comment by dz4news** [comment]
well this was not expected... 
`warning
Timestamp
2025-01-26 16:04:00
Tag
frigate.ptz.onvif
Message
Camera front_ptz does not support the ONVIF GetStatus method. Autotracking will not function correctly and must be disabled in your config.`

The camera is using ONVIF Version 23.06(V3.1.0.1664296)

**Reply by hawkeye217** [reply]
That's definitely surprising to me.

Can you enable debug logs for onvif, restart Frigate, and paste the initial output of the logs after restarting here?

```
logger:
  default: info
  logs:
    frigate.ptz.onvif: debug
```

**Reply by dz4news** [reply]
I sure can... you prefer I open a support ticket so it can be tracked for the model of camera?

**Reply by hawkeye217** [reply]
Yes, go ahead and open your own support discussion. Thanks!

**Reply by hyacin75** [reply]
> well this was not expected... `warning Timestamp 2025-01-26 16:04:00 Tag frigate.ptz.onvif Message Camera front_ptz does not support the ONVIF GetStatus method. Autotracking will not function correctly and must be disabled in your config.`

I'm getting this too on my new SD3C405DB-GNY-A-PV I bought specifically to use with Frigate :'( (and this thread is basically the only result web searching for the error)

Should I do the same debug thing @hawkeye217 ?  Or is it a known issue with this model or anything?

**Reply by hawkeye217** [reply]
Frigate 0.16 (currently in beta) may better indicate why you are seeing this error message.

**Reply by hyacin75** [reply]
> Frigate 0.16 (currently in beta) may better indicate why you are seeing this error message.

Oddly, I'm not getting that message or any complaints about tracking in a couple hours of running 0.16.0-beta3-tensorrt so far.  Though it's still inside for 'testing' so it hasn't really tried to track much, if anything.  I'll keep you posted!  ty!!
